### DISCUSSION TITLE
Is it correct, that ring-* does not work without the base layer in use

### DISCUSSION THREAD

**Discussion by schenktobi** [tailwindlabs/tailwindcss#11506] [Help] [answered]
I use Tailwind for a Chrome Extension, which is injected using an iFrame. I therefore cannot use the base layer, since it would change the look of the Websites the User opens, when the Extension is injected.

When i remove the base layer, most of the utilities still work, but ring just does not work. I created this example in tailwind Play. If you remove `@tailwind base` from the css, the ring-property does not work anymore.

Example: https://play.tailwindcss.com/buCn0mnb0I?file=css

Any help if greatly appreciated and if  I missed something obvious im sorry :D

**Comment by wongjn** [comment, marked as answer]
`ring-*` classes need the base `@layer` to function properly. You may be able to keep the `@tailwind base` rule while preventing the look of websites the user opens by [disabling the Preflight core plugin](https://tailwindcss.com/docs/preflight#disabling-preflight) instead.

**Reply by schenktobi** [reply]
That would be helpful. I will Test it asap. Thank you

**Reply by schenktobi** [reply]
Seems like it is already disabled, but overwrites to existing styles still happens :C. Thank you a lot regardless

**Reply by mybearworld** [reply]
@schenktobi That should not happen. What classes get overridden, specifically?

**Reply by schenktobi** [reply]
I don't know, which classes exactly get overwritten, but for example the github search text-color is messed up as weel as e.g. the search Bar in Heroicons. I can provide Screenshots or try to find out the exact class names if needed.

**Reply by mybearworld** [reply]
That really shouldn't happen. Could you share your tailwindconfig?

**Reply by schenktobi** [reply]
Of course. The Tailwind Config looks like the following. Let me know if I can provide any additional Information. I will try to look into this Thread on a daily basis.

```
/** @type {import('tailwindcss').Config} */
/* eslint-disable */
module.exports = {
  prefix: "syn-",
  content: [
    "./index.html",
    "./src/**/*.{vue,js,ts,jsx,tsx}",
    "./node_modules/flowbite/**/*.js",
  ],
  theme: {
    extend: {
      colors: {
        primary: "#05003F",
        secondary: "#E1EFFE",
        light: "#F9FAFB",
        muted: "#9CA3AF",
        gradientFrom: "#05003F",
        gradientTo: "#079bb0",
        main: {
          500: "#3F83F8",
        },
      },
    },
  },
  plugins: [require("flowbite/plugin")],
  corePlugins: {
    preflight: false,
  },
};

```

**Reply by mybearworld** [reply]
The issue is with the Flowbite plugin. It generates some base rules.

**Reply by schenktobi** [reply]
Well thats good to know. I will try to remove flowbite and test it again. I will let you know. Thank you in advance tho :D
